mantidimaging/core/fitting/bounding_box.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (C) 2021 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+import matplotlib.pyplot as plt
+import numpy as np
+
+from mantidimaging.core.data import ImageStack
+from mantidimaging.core.utility.sensible_roi import SensibleROI
+
+
+def get_bounding_box(stack: ImageStack) -> SensibleROI:
+    y_tol = 0.02
+    x_tol = 0.02
+    data = stack.data[:, 5:-5, 5:-5]  # clip dark edges
+    all_col_mean = np.mean(data, axis=(1, 0))
+    all_row_mean = np.mean(data, axis=(2, 0))
+    all_col_mean_flipped = np.flip(all_col_mean)
+    all_row_mean_flipped = np.flip(all_row_mean)
+
+    plt.plot(np.diff(all_row_mean) / all_row_mean[1:])
+    plt.show()
+
+    def find_bound(data_temp, tol, dim=0):
+        data_temp = data_temp[0:round(len(data_temp) / 2)]
+        while True:
```

Comment:
Can this be done without a potentially infinite loop?

Could `tol` be set based on the range of `np.diff(data_temp)`?